src/signals/incident-management/containers/Dashboard/components/Filter/SelectList.tsx

```diff
@@ -0,0 +1,136 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (C) 2023 Gemeente Amsterdam
+import type { Dispatch, SetStateAction } from 'react'
+import { useCallback, useEffect, useRef } from 'react'
+
+import { ChevronDown } from '@amsterdam/asc-assets'
+import { isNumber } from 'lodash'
+import { useFormContext } from 'react-hook-form'
+
+import OptionsList from './OptionsList'
+import {
+  InvisibleButton,
+  OptionListContainer,
+  RefreshIcon,
+  Select,
+  SelectContainer,
+} from './styled'
+import type { Filter } from './types'
+import { useFilters } from './useFilter'
+
+type Props = {
+  filterActiveName: string
+  setFilterActiveName: Dispatch<SetStateAction<string>>
+}
+
+const SelectList = ({ filterActiveName, setFilterActiveName }: Props) => {
+  const { getValues, reset, setValue, watch } = useFormContext()
+
+  const selectedDepartment = watch('department')
+
+  const filters = useFilters(selectedDepartment)
+
+  // set initial department value
+  useEffect(() => {
+    if (!selectedDepartment.value && filters[0].options[0]) {
+      setValue('department', filters[0].options[0])
+    }
+  }, [filters, selectedDepartment, setValue])
+
+  const chopFilterName = (filterName: string) =>
```

Comment:
Personally I would put this in a utils folder, add a test, and rename it to something with "truncate". "Chop" is not very descriptive. Also, in the test you can add descriptive names for the test so you know why the names need to be truncated in the first place.

Reply:
I can explain my considerations for not extracting this oneliner.